employee-rostering-backend/pom.xml

```diff
@@ -98,6 +96,12 @@
     <dependency>
       <groupId>org.springframework.boot</groupId>
       <artifactId>spring-boot-starter-web</artifactId>
+      <exclusions>
+        <exclusion>
+          <groupId>commons-logging</groupId>
+          <artifactId>commons-logging</artifactId>
```

Comment:
This has no effect. `spring-boot-starter-web` doesn't depend on `commons-logging` (not even transitively).

Reply:
Oddly enough, if included in employee-rostering-distribution, it suddenly becomes a transitive dependency and maven enforcer complains. (I have no idea why this is the case)

Reply:
Can you share the output of
```sh
mvn dependency\:tree -Denforcer.skip -B
```
ran in `employee-rostering-distribution` when you get the enforcer complaint? (With the exclusion removed).

Reply:
Because I don't see it when trying it locally. It's strange.

Reply:
```
[INFO] org.optaweb.employeerostering:employee-rostering-distribution:jar:7.28.0-SNAPSHOT
[INFO] +- org.optaweb.employeerostering:employee-rostering-frontend:jar:7.28.0-SNAPSHOT:compile
[INFO] +- org.optaweb.employeerostering:employee-rostering-backend:jar:7.28.0-SNAPSHOT:compile
[INFO] |  +- org.springframework.boot:spring-boot-starter-web:jar:2.1.8.RELEASE:compile
[INFO] |  |  +- org.springframework.boot:spring-boot-starter:jar:2.1.8.RELEASE:compile
[INFO] |  |  |  +- org.springframework.boot:spring-boot-starter-logging:jar:2.1.8.RELEASE:compile
[INFO] |  |  |  |  +- org.apache.logging.log4j:log4j-to-slf4j:jar:2.11.2:compile
[INFO] |  |  |  |  |  \- org.apache.logging.log4j:log4j-api:jar:2.11.2:compile
[INFO] |  |  |  |  \- org.slf4j:jul-to-slf4j:jar:1.7.26:compile
[INFO] |  |  |  +- javax.annotation:javax.annotation-api:jar:1.3.2:compile
[INFO] |  |  |  +- org.springframework:spring-core:jar:4.3.3.RELEASE:compile
**[INFO] |  |  |  |  \- commons-logging:commons-logging:jar:1.1.1:compile**
[INFO] |  |  |  \- org.yaml:snakeyaml:jar:1.18:runtime
[INFO] |  |  +- org.springframework.boot:spring-boot-starter-json:jar:2.1.8.RELEASE:compile
[INFO] |  |  |  +- com.fasterxml.jackson.datatype:jackson-datatype-jdk8:jar:2.9.9:compile
[INFO] |  |  |  +- com.fasterxml.jackson.datatype:jackson-datatype-jsr310:jar:2.9.9:compile
[INFO] |  |  |  \- com.fasterxml.jackson.module:jackson-module-parameter-names:jar:2.9.9:compile
[INFO] |  |  +- org.springframework.boot:spring-boot-starter-tomcat:jar:2.1.8.RELEASE:compile
[INFO] |  |  |  +- org.apache.tomcat.embed:tomcat-embed-core:jar:9.0.24:compile
[INFO] |  |  |  +- org.apache.tomcat.embed:tomcat-embed-el:jar:9.0.24:compile
[INFO] |  |  |  \- org.apache.tomcat.embed:tomcat-embed-websocket:jar:9.0.24:compile
[INFO] |  |  +- org.hibernate.validator:hibernate-validator:jar:6.0.13.Final:compile
[INFO] |  |  |  \- org.jboss.logging:jboss-logging:jar:3.3.2.Final:compile
[INFO] |  |  +- org.springframework:spring-web:jar:4.3.3.RELEASE:compile
[INFO] |  |  |  +- org.springframework:spring-aop:jar:4.3.3.RELEASE:compile
[INFO] |  |  |  +- org.springframework:spring-beans:jar:4.3.3.RELEASE:compile
[INFO] |  |  |  \- org.springframework:spring-context:jar:4.3.3.RELEASE:compile
[INFO] |  |  \- org.springframework:spring-webmvc:jar:4.3.3.RELEASE:compile
[INFO] |  |     \- org.springframework:spring-expression:jar:4.3.3.RELEASE:compile
[INFO] |  +- org.springframework.boot:spring-boot-starter-data-jpa:jar:2.1.8.RELEASE:compile
[INFO] |  |  +- org.springframework.boot:spring-boot-starter-aop:jar:2.1.8.RELEASE:compile
[INFO] |  |  |  \- org.aspectj:aspectjweaver:jar:1.9.4:compile
[INFO] |  |  +- org.springframework.boot:spring-boot-starter-jdbc:jar:2.1.8.RELEASE:compile
[INFO] |  |  |  +- com.zaxxer:HikariCP:jar:3.2.0:compile
[INFO] |  |  |  \- org.springframework:spring-jdbc:jar:4.3.3.RELEASE:compile
[INFO] |  |  +- javax.transaction:javax.transaction-api:jar:1.3:compile
[INFO] |  |  +- javax.xml.bind:jaxb-api:jar:2.3.1:compile
[INFO] |  |  |  \- javax.activation:javax.activation-api:jar:1.2.0:compile
[INFO] |  |  +- org.hibernate:hibernate-core:jar:5.3.7.Final:compile
[INFO] |  |  |  +- javax.persistence:javax.persistence-api:jar:2.2:compile
[INFO] |  |  |  +- org.javassist:javassist:jar:3.22.0-GA:compile
[INFO] |  |  |  +- antlr:antlr:jar:2.7.7:compile
[INFO] |  |  |  +- org.jboss:jandex:jar:2.1.1.Final:compile
[INFO] |  |  |  +- org.dom4j:dom4j:jar:2.1.1:compile
[INFO] |  |  |  \- org.hibernate.common:hibernate-commons-annotations:jar:5.0.4.Final:compile
[INFO] |  |  +- org.springframework.data:spring-data-jpa:jar:2.1.10.RELEASE:compile
[INFO] |  |  |  +- org.springframework.data:spring-data-commons:jar:2.1.10.RELEASE:compile
[INFO] |  |  |  +- org.springframework:spring-orm:jar:4.3.3.RELEASE:compile
[INFO] |  |  |  \- org.springframework:spring-tx:jar:4.3.3.RELEASE:compile
[INFO] |  |  \- org.springframework:spring-aspects:jar:4.3.3.RELEASE:compile
[INFO] |  +- com.h2database:h2:jar:1.3.173:runtime
[INFO] |  +- org.springframework.boot:spring-boot-devtools:jar:2.1.8.RELEASE:runtime
[INFO] |  |  +- org.springframework.boot:spring-boot:jar:2.1.8.RELEASE:compile
[INFO] |  |  \- org.springframework.boot:spring-boot-autoconfigure:jar:2.1.8.RELEASE:compile
[INFO] |  +- org.springframework.boot:spring-boot-configuration-processor:jar:2.1.8.RELEASE:compile
[INFO] |  +- org.optaplanner:optaplanner-core:jar:7.28.0-SNAPSHOT:compile
[INFO] |  |  +- org.kie:kie-api:jar:7.28.0-SNAPSHOT:compile
[INFO] |  |  |  \- org.kie.soup:kie-soup-maven-support:jar:7.28.0-SNAPSHOT:compile
[INFO] |  |  +- org.kie:kie-internal:jar:7.28.0-SNAPSHOT:compile
[INFO] |  |  +- org.drools:drools-core:jar:7.28.0-SNAPSHOT:compile
[INFO] |  |  |  +- org.mvel:mvel2:jar:2.4.4.Final:compile
[INFO] |  |  |  +- org.kie.soup:kie-soup-commons:jar:7.28.0-SNAPSHOT:compile
[INFO] |  |  |  +- org.kie.soup:kie-soup-project-datamodel-commons:jar:7.28.0-SNAPSHOT:compile
[INFO] |  |  |  |  \- org.kie.soup:kie-soup-project-datamodel-api:jar:7.28.0-SNAPSHOT:compile
[INFO] |  |  |  \- commons-codec:commons-codec:jar:1.10:compile
[INFO] |  |  +- org.drools:drools-compiler:jar:7.28.0-SNAPSHOT:compile
[INFO] |  |  |  +- org.antlr:antlr-runtime:jar:3.5.2:compile
[INFO] |  |  |  +- org.eclipse.jdt.core.compiler:ecj:jar:4.6.1:compile
[INFO] |  |  |  \- com.google.protobuf:protobuf-java:jar:3.6.1:compile
[INFO] |  |  +- org.apache.commons:commons-lang3:jar:3.8.1:compile
[INFO] |  |  +- org.apache.commons:commons-math3:jar:3.4.1:compile
[INFO] |  |  +- com.google.guava:guava:jar:25.0-jre:compile
[INFO] |  |  |  +- com.google.code.findbugs:jsr305:jar:1.3.9:compile
[INFO] |  |  |  +- org.checkerframework:checker-compat-qual:jar:2.0.0:compile
[INFO] |  |  |  +- com.google.errorprone:error_prone_annotations:jar:2.1.3:compile
[INFO] |  |  |  +- com.google.j2objc:j2objc-annotations:jar:1.1:compile
[INFO] |  |  |  \- org.codehaus.mojo:animal-sniffer-annotations:jar:1.14:compile
[INFO] |  |  +- org.reflections:reflections:jar:0.9.11:compile
[INFO] |  |  \- com.thoughtworks.xstream:xstream:jar:1.4.11.1:compile
[INFO] |  |     +- xmlpull:xmlpull:jar:1.1.3.1:compile
[INFO] |  |     \- xpp3:xpp3_min:jar:1.1.4c:compile
[INFO] |  +- org.optaplanner:optaplanner-benchmark:jar:7.28.0-SNAPSHOT:compile
[INFO] |  |  +- org.optaplanner:optaplanner-persistence-common:jar:7.28.0-SNAPSHOT:compile
[INFO] |  |  +- org.optaplanner:optaplanner-persistence-xstream:jar:7.28.0-SNAPSHOT:compile
[INFO] |  |  +- org.optaplanner:optaplanner-persistence-jaxb:jar:7.28.0-SNAPSHOT:compile
[INFO] |  |  |  +- org.jboss.spec.javax.xml.bind:jboss-jaxb-api_2.3_spec:jar:1.0.1.Final:compile
[INFO] |  |  |  +- com.sun.xml.bind:jaxb-core:jar:2.3.0:compile
[INFO] |  |  |  \- com.sun.xml.bind:jaxb-impl:jar:2.3.0:compile
[INFO] |  |  +- commons-io:commons-io:jar:2.6:compile
[INFO] |  |  +- org.jfree:jfreechart:jar:1.0.19:compile
[INFO] |  |  +- org.jfree:jcommon:jar:1.0.23:compile
[INFO] |  |  \- org.freemarker:freemarker:jar:2.3.26.jbossorg-1:compile
[INFO] |  +- org.optaplanner:optaplanner-persistence-jackson:jar:7.28.0-SNAPSHOT:compile
[INFO] |  |  +- com.fasterxml.jackson.core:jackson-core:jar:2.9.9:compile
[INFO] |  |  \- com.fasterxml.jackson.core:jackson-databind:jar:2.9.9.3:compile
[INFO] |  |     \- com.fasterxml.jackson.core:jackson-annotations:jar:2.9.9:compile
[INFO] |  +- net.bytebuddy:byte-buddy-dep:jar:1.9.12:compile
[INFO] |  |  +- org.ow2.asm:asm:jar:5.0.4:compile
[INFO] |  |  \- org.ow2.asm:asm-commons:jar:7.1:compile
[INFO] |  +- javax.validation:validation-api:jar:1.0.0.GA:compile
[INFO] |  +- io.springfox:springfox-swagger2:jar:2.9.2:compile
[INFO] |  |  +- io.swagger:swagger-annotations:jar:1.5.20:compile
[INFO] |  |  +- io.swagger:swagger-models:jar:1.5.20:compile
[INFO] |  |  +- io.springfox:springfox-spi:jar:2.9.2:compile
[INFO] |  |  |  \- io.springfox:springfox-core:jar:2.9.2:compile
[INFO] |  |  +- io.springfox:springfox-schema:jar:2.9.2:compile
[INFO] |  |  +- io.springfox:springfox-swagger-common:jar:2.9.2:compile
[INFO] |  |  +- io.springfox:springfox-spring-web:jar:2.9.2:compile
[INFO] |  |  +- com.fasterxml:classmate:jar:1.4.0:compile
[INFO] |  |  +- org.springframework.plugin:spring-plugin-core:jar:1.2.0.RELEASE:compile
[INFO] |  |  +- org.springframework.plugin:spring-plugin-metadata:jar:1.2.0.RELEASE:compile
[INFO] |  |  \- org.mapstruct:mapstruct:jar:1.2.0.Final:compile
[INFO] |  +- io.springfox:springfox-swagger-ui:jar:2.9.2:compile
[INFO] |  +- org.slf4j:slf4j-api:jar:1.7.26:compile
[INFO] |  \- ch.qos.logback:logback-classic:jar:1.2.3:compile
[INFO] |     \- ch.qos.logback:logback-core:jar:1.2.3:compile
[INFO] +- org.optaweb.employeerostering:employee-rostering-backend:jar:exec:7.28.0-SNAPSHOT:compile
[INFO] +- org.optaweb.employeerostering:employee-rostering-docs:zip:7.28.0-SNAPSHOT:compile
[INFO] +- junit:junit:jar:4.12:test
[INFO] |  \- org.hamcrest:hamcrest-core:jar:1.3:test
[INFO] \- org.mockito:mockito-core:jar:2.12.0:test
[INFO]    +- net.bytebuddy:byte-buddy:jar:1.9.4:compile
[INFO]    +- net.bytebuddy:byte-buddy-agent:jar:1.7.9:test
[INFO]    \- org.objenesis:objenesis:jar:2.1:test

```

Reply:
Thanks, I finally found out what's going on. I probably made some kind of stupid mistake due to which I couldn't see this on my machine.

The problem: You have a "very" old version of `org.springframework:*` artifacts in your dependency tree. The old `spring-core` [depends](https://mvnrepository.com/artifact/org.springframework/spring-core/4.3.3.RELEASE) on `commons-logging` but a more recent version [does not](https://mvnrepository.com/artifact/org.springframework/spring-core/5.1.9.RELEASE).

This is a bit unfortunate situation caused by the fact that kie-parent sets spring (framework) artifacts version in its [dependency management](https://github.com/kiegroup/droolsjbpm-build-bootstrap/search?q=springframework&unscoped_q=springframework) and the version hasn't been updated for a long time. This is *bad* because spring framework artifact versions (among others) are supposed to be managed by [spring-boot-dependencies](https://github.com/spring-projects/spring-boot/blob/v2.1.8.RELEASE/spring-boot-project/spring-boot-dependencies/pom.xml#L2762-L2768) (which imports [spring-framework-bom](https://mvnrepository.com/artifact/org.springframework/spring-framework-bom/5.1.9.RELEASE)). I solve this in [vehicle routing by overriding](https://github.com/kiegroup/optaweb-vehicle-routing/blob/7.27.0.Final/pom.xml#L45-L46) the `version.org.springframework` property. We have to manually make sure that its value is same as the `spring-boot-dependencies` [BOM uses](https://github.com/spring-projects/spring-boot/blob/v2.1.8.RELEASE/spring-boot-project/spring-boot-dependencies/pom.xml#L168). This is not nice but it's a good workaround. I think that kie-parent managing spring dependencies is a bad idea. We can try and fix it later.

Please override the `version.org.springframework` property.

Reply:
Thanks for tracking down that issue; fixed